chartfx-chart/src/main/java/de/gsi/chart/axes/spi/AbstractAxisParameter.java

```diff
@@ -1115,200 +948,372 @@ public void setAutoRangeRounding(final boolean round) {
-        minorTickLengthProperty().set(value);
+    public DoubleProperty minorTickLengthProperty() {
+        return minorTickLength;
+    }
+
+    public double getMinorTickLength() {
+        return minorTickLengthProperty().get();
+    }
+
+    public void setMinorTickLength(final double value) {
+        minorTickLengthProperty().set(value);
+    }
+
+    public BooleanProperty minorTickVisibleProperty() {
+        return minorTickVisible;
+    }
+
+    public boolean isMinorTickVisible() {
+        return minorTickVisibleProperty().get();
     }
 
     public void setMinorTickVisible(final boolean value) {
         minorTickVisibleProperty().set(value);
     }
 
     @Override
-    public void setName(final String value) {
-        nameProperty().set(value);
+    public DoubleProperty minProperty() {
+        return minProp;
     }
 
-    public void setOverlapPolicy(final AxisLabelOverlapPolicy value) {
-        overlapPolicyProperty().set(value);
+    @Override
+    public double getMin() {
+        return minProperty().get();
     }
 
     @Override
-    public void setSide(final Side value) {
-        sideProperty().set(value);
+    public boolean setMin(final double value) {
+        final double oldvalue = minProperty().get();
+        minProperty().set(value);
+        return oldvalue != value;
     }
 
-    public void setTickLabelFill(final Paint value) {
-        tickLabelFillProperty().set(value);
+    @Override
+    public StringProperty nameProperty() {
+        return axisName;
     }
 
-    public void setTickLabelFont(final Font value) {
-        tickLabelFontProperty().set(value);
+    @Override
+    public String getName() {
+        return nameProperty().get();
     }
 
-    public void setTickLabelFormatter(final StringConverter<Number> value) {
-        tickLabelFormatterProperty().setValue(value);
+    @Override
+    public void setName(final String value) {
+        nameProperty().set(value);
     }
 
-    public void setTickLabelGap(final double value) {
-        tickLabelGapProperty().set(value);
+    public ObjectProperty<AxisLabelOverlapPolicy> overlapPolicyProperty() {
+        return overlapPolicy;
     }
 
-    public void setTickLabelRotation(final double value) {
-        tickLabelRotationProperty().setValue(value);
+    public AxisLabelOverlapPolicy getOverlapPolicy() {
+        return overlapPolicyProperty().get();
     }
 
-    public void setTickLabelsVisible(final boolean value) {
-        tickLabelsVisibleProperty().set(value);
+    public void setOverlapPolicy(final AxisLabelOverlapPolicy value) {
+        overlapPolicyProperty().set(value);
     }
 
-    public void setTickLength(final double value) {
-        tickLengthProperty().set(value);
+    public ReadOnlyDoubleProperty scaleProperty() {
+        return scale.getReadOnlyProperty();
     }
 
-    public void setTickMarkVisible(final boolean value) {
-        tickMarkVisibleProperty().set(value);
+    ReadOnlyDoubleWrapper scalePropertyImpl() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5110477)